modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -316,6 +333,9 @@ TensorMechanicsAction::actSubdomainChecks()
         mooseError("The TensorMechanics action requires all subdomains to have the same coordinate "
                    "system.");
   }
+
+  if (_coord_system == Moose::COORD_RZ && _out_of_plane_direction != OutOfPlaneDirection::z)
```

Comment:
Looks like _coord_system isn't set in the constructor. You didn't put this check inside of the "validate_coordinate_systems" tasks block which is why you are failing valgrind now.

Reply:
I've corrected this.  Since it is merged do I just make another merge request that references the original issue and address some of your other comments.